frontend/packages/ceph-storage-plugin/src/components/modals/clone-pvc-modal/clone-pvc-modal.tsx

```diff
@@ -0,0 +1,190 @@
+        setButton(true);
+        throw error;
+      });
+  };
+
+  const updateClonePVCName: React.ReactEventHandler<HTMLInputElement> = (event) => {
+    setClonePVCName(event.currentTarget.value);
+  };
+
+  return (
+    <form onSubmit={submit} className="modal-content modal-content--no-inner-scroll">
+      <ModalTitle>Clone</ModalTitle>
+      <ModalBody>
+        <div className="form-group">
+          <label htmlFor="clone-pvc-modal__name" className="co-required">
+            Name
+          </label>
+          <input
+            className="pf-c-form-control"
+            name="pvcName"
+            id="clone-pvc-modal__name"
+            type="text"
+            onChange={updateClonePVCName}
+            value={clonePVCName}
+            required
+          />
+        </div>
+        <div className="form-group clone-pvc-modal__details-section">
+          <p className="text-muted clone-pvc-modal__pvc-details">PVC Details</p>
+          <div className="clone-pvc-modal__details">
+            <div>
+              <div>
+                <p className="clone-pvc-modal__pvc-details">Namespace</p>
+                <p>
+                  <ResourceIcon kind="Namespace" />
+                  {resource.metadata.namespace}
+                </p>
+              </div>
+              <div>
+                <p className="clone-pvc-modal__pvc-details">Storage Class</p>
+                <p>
+                  <ResourceIcon kind="StorageClass" />
+                  {resource.spec.storageClassName}
+                </p>
+              </div>
+            </div>
+            <div>
+              <div>
+                <p className="clone-pvc-modal__pvc-details">Requested Capacity</p>
+                <p>
+                  {
+                    humanizeBinaryBytes(
+                      convertToBaseValue(resource.spec.resources.requests.storage),
+                    ).string
+                  }
+                </p>
+              </div>
+              <div>
+                <p className="clone-pvc-modal__pvc-details">Used Capacity</p>
+                <p>{!queriesLoadError && humanizeBinaryBytes(pvcUsedCapacity).string}</p>
+              </div>
+            </div>
+            <div>
+              <div>
+                <p className="clone-pvc-modal__pvc-details">Access Mode</p>
+                <p>{accessModeRadios[resource.spec.accessModes]}</p>
+              </div>
+              <div>
+                <p className="clone-pvc-modal__pvc-details">Volume Mode</p>
+                <p>{resource.spec.volumeMode}</p>
+              </div>
+            </div>
+          </div>
+        </div>
+      </ModalBody>
+      <ModalSubmitFooter
+        inProgress={inProgress}
+        errorMessage={errorMessage}
+        submitText="Clone"
+        cancel={cancel}
+        submitDisabled={buttonDisabled}
+      />
+    </form>
+  );
+});
+
+export type ClonePVCModalProps = {
+  kind?: any;
+  handlePromise: <T>(promise: Promise<T>) => Promise<T>;
```

Comment:
not required. `handlePromiseProps` should have this

Reply:
ack